docs/proposals/20190503-TalosNodeStatus.md

```diff
@@ -0,0 +1,49 @@
+# Proposal: [Title]
+
+Author(s): [Andrew Rynhard](@andrewrynhard) [Brad Beam](@bradbeam)
+
+Last updated: May 3 2019
+
+## Abstract
+
+Without insight into the current status/health of a Talos node, we severely limit ourselves in the quality and dependability of any automation we implement.
+We propose that we need an API exposed on each node that gives detailed information about the nodes current status/heatlh.
+A common set of data between masters and workers will be implemented along with data specific to masters that we can make use of in other features, like upgrades.
+
+## Background
+
+We have long thought that in general a status API would be useful.
+Once we started to desgin upgrades it became clear that we needed an API for node status/heath that we could use to orchestrate automated, cluster-wide upgrades.
+
+## Proposal
+
+### Technology
+
+We propose that we leverage the existing `osd` service and extend it to include the following APIs.
+
+### Common Status API
+
+The common status API is a set of metrics we require every node to have.
```

Comment:
I don't know if that would be enough or not, but this status API could be implemented via service healthchecks (depends on what kind of health checks we implement), but this could be for example: "kubeadm" finished without errors (or skipped), containerd running & healthy (responds to 'health' API), kubelet healthy (idk what can be checked?)

Reply:
I definitely think that will contribute to what we consider healthy.

- All services are running
- Can query Kubernetes API form local machine (maybe only for masters)

Reply:
> Can query Kubernetes API form local machine (maybe only for masters)

I wonder if that could be kubelet health definition for masters?

Reply:
possibly ... but some times Kubelet running isn't enough... maybe we extend it to kubelet + all control plane components